keyboards/yellowbrickroad/keymaps/led/keymap.c

```diff
@@ -0,0 +1,1002 @@
+    [_BASEANDCHORDS]     = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_HEPTATONIC]        = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_PENTATONIC]        = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_TRANS]             = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(MI_TRSD, MI_TRSU) },
+    [_FLIP]              = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_FLIPCHORDS]        = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_FLIPBASEANDCHORDS] = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_FLIPHEPTA]         = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_FLIPPENTA]         = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(_______, _______)  },
+    [_FLIPTRANS]         = { ENCODER_CCW_CW(_______,  _______), ENCODER_CCW_CW(MI_TRSU, MI_TRSD) },
+    [_FN]                = { ENCODER_CCW_CW(RGB_RMOD, RGB_MOD), ENCODER_CCW_CW(MI_OCTD, MI_OCTU)  }
+};
+#endif
+
+
+void reset_scale_indicator(void) {
+    //  reset transpose value and scale_indicator_col to default.
+    midi_config.transpose = 0;
+    scale_indicator_col = DEFAULT_SCALE_COL;
+}
+
+void my_init(void){
+    //  Set octave to QK_MIDI_OCTAVE_0
+    midi_config.octave = QK_MIDI_OCTAVE_0 - MIDI_OCTAVE_MIN;
+    // avoid using 127 since it is used as a special number in some sound sources.
+    midi_config.velocity = MIDI_INITIAL_VELOCITY;
+
+    reset_scale_indicator();
+    is_trans_mode = false;      //  trans mode is disabled by default.
+}
+
+void _load_eeprom_data(void){
+    // no need to load isSingleBass since it's called directly by IS_SINGLE_BASS() macro.
+    midi_chord_ch = user_config.chord_ch;
+    midi_bass_ch = user_config.bass_ch;
+}
+
+void eeconfig_init_user(void) {  // EEPROM is getting reset!
+    midi_init();  // in midi_init(), octave is set to MI_OCT_2.
+    my_init();    // Octave is set to QK_MIDI_OCTAVE_0, velocity is set to MIDI_INITIAL_VELOCITY. etc.
+
+    // UNISON flags
+    melody_dyad_high = false;  //  true when +1 octave unison dyad is enabled.
+    melody_dyad_low  = false;  //  true when -1 octave unison dyad is enabled.
+    melody_unison_suppress  = true;  //  true: velocity of octave unison note is suppressd to UNISON_VELOCITY_RATIO
+
+    //  Reset Bass setting
+    user_config.raw = 0;  // default: dyad, chord_ch and bass_ch are DEFAULT_MAIN_CH_NUMBER.
+    eeconfig_update_user(user_config.raw);
+    _load_eeprom_data();
+
+    //  Reset the midi keyboard layout
+    default_layer_set(_LS_BASE);
+    layer_state_set(_LS_BASE);
+    current_default_layer = _BASE;
+    layerFlipMode = false;
+
+#ifdef RGB_MATRIX_ENABLE
+    rgb_matrix_sethsv(HSV_BLUE);
+    //  party mode (for LED soldering test.)
+    rgb_matrix_mode(RGB_MATRIX_RAINBOW_MOVING_CHEVRON);
```

Comment:
There are actually defines to set the default here, and this isn't needed

Reply:
These statements are deleted, and the defines in led/config.h are added.